File: solutions/005/005.c
#include <stdio.h>

int gcd(int a, int b) {
    int c;
    while (b != 0) {
        c = b;
        b = a % b;
        a = c;
    }
    return a;
}

int lcm(int a, int b) {
    return a * b / gcd(a, b);
}

int main() {
    int result = 1;
    for (int i = 1; i < 20; i++)
        result = lcm(result, i);
    printf("%d\n", result);
}

File: solutions/005/005.rs
fn gcd(mut a : i32 , mut b : i32) -> i32 {
	while b != 0 {
        let c = b;
        b = a % b;
        a = c;
	}
	return a
}

fn lcm(a : i32, b : i32) -> i32 {
    return a * b / gcd(a, b);
}

fn main() {
    let mut result = 1;
    for i in 1..20 {
        result = lcm(result, i);
    }
    println!("{}", result);
}
